Add 'Y-m-d H:i:s' support to DateTimeImmutableFactory

craig410 opened the pull request.

craig410 committed:
Add 'Y-m-d H:i:s' support to DateTimeImmutableFactory

acoulton commented:
I think we wanted this to be strict, so require a (not nullable) string & throw if empty or invalid, rather than returning the InvalidUserDatTime class?

Also, probably want to assert to full precision (including micros & timezone) - createFromFormat I think backfills from current time by default if there's components that aren't specified in the format string - see the tests for the ::atMicrotime method.

craig410 committed:
Deprecate DateTimeFactory methods which return an InvalidUserDateTime

There are better patterns than this, user input should be validated long before getting to the point of creating a DateTime and this handled client-side

craig410 commented:
@acoulton happy I release this?

acoulton reviewed (approved):
Awesome go for it

craig410 committed:
Bump changelog for release
